plutus-tx-plugin/src/PlutusTx/Compiler/Types.hs

```diff
@@ -162,6 +162,7 @@ type Compiling uni fun m =
     , MonadWriter CoverageIndex m
     , PLC.GShow uni, PLC.GEq uni
     , PLC.ToBuiltinMeaning uni fun
+    , uni `PLC.Contains` ()
```

Comment:
Hopefully, this is not fighting a hack with another hack... Personally, I think it's fine demanding a unit for any variant.

Reply:
This is partly what `CompilingDefault` is for, although that's a bit overkill in this case. For now I'd say just use `CompilingDefault` in `compileTyCon` and I can worry about chopping up `CompilingDefault` at some point.

Reply:
The `CompilingDefault` change is more pervasive but looking at the surrounding code it is indeed better. I have updated as suggested.